the-collab-lab/tcl-32-smart-shopping-list
# Adding Search Functionality to search list for items

## Description

This PR adds SearchList.js as a child component to List.js. The code adds a search field that when typed in will search through the list and display only items that contain the search parameters. Once typing, an "X" will display and when clicked, will clear the search box and show the full list again.

The search algorithm is using the Helper.js function to search with a normalized value using a filter. 

## Related Issue

Closes #9 

## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Updates

### Before

![](https://cdn.zappy.app/5494c138dc3f5a80e8d165899bc1f6d4.png)


### After

![](https://cdn.zappy.app/0de616e13bd2b8d5f0c318bd35b42257.gif)


## Testing Steps / QA Criteria

1. ```git checkout sk-jr-search-list```
2. ```npm start```
3. Create new list or add token from existing list 
4. Use search box to type out what you are looking for
5. Verify it searches and displays only items that contain the search 
6. Click "X" to clear your search and see the full list again
